### PR TITLE
Synchronize tests

### PR DESCRIPTION
**Pull request by lukechampine**

### CODE REVIEW

File: modules/consensus/synchronize_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+	// Mine on cst2 until it is more than 'MaxCatchUpBlocks' ahead of cst2.
+	// NOTE: we have to disconnect prior to this, otherwise cst2 will relay
+	// blocks to cst1.
+	err = cst1.gateway.Disconnect(cst2.gateway.Address())
+	if err != nil {
+		t.Fatal(err)
+	}
+	// TODO: more than 30 causes a race condition!
+	for cst2.cs.Height() < cst1.cs.Height()+20 {
```
Comment (lukechampine) [inline modules/consensus/synchronize_test.go]: this seems to be non-deterministic.

File: modules/consensus/synchronize_test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	cst2.cs.mu.Unlock(lockID)
+
+	err = cst1.cs.Synchronize(cst2.gateway.Address())
+	if err != ErrBlockKnown {
+		t.Fatal("expected ErrBlockKnown, got", err)
+	}
+}
+
+func TestResynchronize(t *testing.T) {
+	t.Skip("takes way too long")
```
Comment (lukechampine) [inline modules/consensus/synchronize_test.go]: "too long" meaning 3 minutes.

**Comment by DavidVorick** [comment]
deadlock in negotiate?

hmm

**Comment by DavidVorick** [comment]
I looked at the code causing the deadlock and.... ???

I'm not sure why that deadlocked. There doesn't seem to be any locking inside of that function anywhere.

**Pull request merged**